src/app/home-page/widget/starred-box/starred-box.component.ts

```diff
@@ -0,0 +1,75 @@
+import { HttpErrorResponse } from '@angular/common/http';
+import { Component, OnInit } from '@angular/core';
+import { AlertService } from 'app/shared/alert/state/alert.service';
+import { Base } from 'app/shared/base';
+import { Dockstore } from 'app/shared/dockstore.model';
+import { Event, UsersService, EventsService } from 'app/shared/openapi';
+import { finalize, takeUntil } from 'rxjs/operators';
+
+@Component({
+  selector: 'app-starred-box',
+  templateUrl: './starred-box.component.html',
+  styleUrls: ['./starred-box.component.scss'],
+})
+export class StarredBoxComponent extends Base implements OnInit {
+  Dockstore = Dockstore;
+  totalStarredWorkflows: number = 0;
+  totalStarredTools: number = 0;
+  totalStarredServices: number = 0;
+  totalStarredOrganizations: number = 0;
+  events: Array<Event> = [];
+  public isLoading = true;
+  EventType = Event.TypeEnum;
+  readonly supportedEventTypes = [
+    Event.TypeEnum.ADDVERSIONTOENTRY,
+    Event.TypeEnum.CREATECOLLECTION,
+    Event.TypeEnum.ADDTOCOLLECTION,
+    Event.TypeEnum.PUBLISHENTRY,
+    Event.TypeEnum.UNPUBLISHENTRY,
+    Event.TypeEnum.MODIFYCOLLECTION,
+    Event.TypeEnum.ADDUSERTOORG,
+  ];
+
+  constructor(private usersService: UsersService, private eventsService: EventsService, private alertService: AlertService) {
+    super();
+  }
+
+  ngOnInit(): void {
+    this.getMyEvents();
+    this.usersService.getStarredServices().subscribe((starredServices) => {
+      this.totalStarredServices = starredServices.length;
+    });
+    this.usersService.getStarredTools().subscribe((starredTools) => {
+      this.totalStarredTools = starredTools.length;
+    });
+    this.usersService.getStarredWorkflows().subscribe((starredWorkflows) => {
+      this.totalStarredWorkflows = starredWorkflows.length;
+    });
+    this.usersService.getStarredOrganizations().subscribe((starredOrganizations) => {
+      this.totalStarredOrganizations = starredOrganizations.length;
+    });
+  }
+
+  getMyEvents() {
+    this.eventsService
+      .getEvents('ALL_STARRED')
+      .pipe(
+        finalize(() => (this.isLoading = false)),
+        takeUntil(this.ngUnsubscribe)
+      )
+      .subscribe(
+        (events) => {
+          events
```

Comment:
I think we chatted about this on another PR, and you can leave it, but you could write this as this, which to me at least, is more readable (but others' opinions may vary):

```
this.events = events.filter((event) => this.supportedEventTypes.includes(event.type))
   .slice(0,4);
```

Also, I'm not sure if the 4 should be a 3 or a 4. :)